contracts/src/mocks/MockFunctionGateway.sol

```diff
@@ -1,4 +1,184 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.16;
 
-contract MockFunctionGateway {}
+import {VmSafe} from "forge-std/Vm.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {IFunctionGateway} from "../interfaces/IFunctionGateway.sol";
+
+/// @title MockFunctionGateway
+/// @notice A Mock version of FunctionGateway for testing.
+/// @dev This contract is only meant to be used in tests. To use it, create a json fixture with keys "input"
+///      and "output". Call loadFixture() with the path, then this contract will automatically fulfill
+///      requests with the output.
+contract MockFunctionGateway is IFunctionGateway {
+    VmSafe private constant vm = VmSafe(address(uint160(uint256(keccak256("hevm cheat code")))));
+    uint32 public nonce;
+    bytes32 public verifiedFunctionId;
+    bytes32 public verifiedInputHash;
+    bytes public verifiedOutput;
+    bool public isCallback;
+    mapping(uint32 => bytes32) public requests;
+    mapping(bytes32 => bytes) public outputs;
+
+    function loadFixture(string memory _path) external {
```

Comment:
This looks awesome! Can we also add another function that's `loadInputOutput(bytes memory input, bytes memory output)` so that people can load stuff inline as well?

Reply:
Added